os/arch/arm/src/imxrt/imxrt_hyperflash.c

```diff
@@ -84,30 +84,29 @@
 /************************************************************************************
  * Pre-processor Definitions
  ************************************************************************************/
-#define IMXRT_FLEXSPI FLEXSPI
-#define IMXRT_FLASH_SIZE 0x10000
-#define IMXRT_FLEXSPI_AMBA_BASE FlexSPI_AMBA_BASE
-#define IMXRT_FLASH_PAGE_SIZE 512
-#define IMXRT_SECTOR_SIZE 0x40000
-#define IMXRT_BLOCK_SIZE 0x40000
-#define IMXRT_FLASH_BASE 0x60000000
-#define IMXRT_FLEXSPI_CLOCK kCLOCK_FlexSpi
-#define IMXRT_FLASH_TOTAL_SIZE (IMXRT_FLASH_SIZE*1024)
-#define HYPERFLASH_CMD_LUT_SEQ_IDX_READDATA 0
-#define HYPERFLASH_CMD_LUT_SEQ_IDX_WRITEDATA 1
-#define HYPERFLASH_CMD_LUT_SEQ_IDX_READSTATUS 2
-#define HYPERFLASH_CMD_LUT_SEQ_IDX_WRITEENABLE 4
-#define HYPERFLASH_CMD_LUT_SEQ_IDX_ERASESECTOR 6
-#define HYPERFLASH_CMD_LUT_SEQ_IDX_PAGEPROGRAM 10
-#define CUSTOM_LUT_LENGTH 48
 
 /************************************************************************************
  * Private Data
  ************************************************************************************/
+FAR struct mtd_dev_s *flash_dev = NULL;
 
 /*******************************************************************************
  * Code
  ******************************************************************************/
+#ifdef CONFIG_MTD_PARTITION
+struct mtd_partition_size_info {
+	enum mtd_partition_tag_s tag;
+	ssize_t size;
+};
+
+struct mtd_partition_size_info psize_info[] = {
```

Comment:
This is flash driver, you don't have to maintain partition info & mtd info.
Please apply only flash operations.